jvm/src/main/scala/cats/jvm/Task.scala

```diff
@@ -0,0 +1,166 @@
+    def newThread(r: Runnable) = {
+      val t = defaultThreadFactory.newThread(r)
+      t.setDaemon(true)
+      t
+    }
+  }
+
+  /**
+   * The default executor service is a fixed thread pool with N daemon threads,
+   * where N is equal to the twice number of available processors.
+   */
+  def createFixedExecutor(numThreads: Int): ExecutorService = {
+    Executors.newFixedThreadPool(numThreads, DaemonThreadFactory)
+  }
+
+
+  /**
+   * Construct a Task which represents an already calculated eager
+   * value
+   */
+  def now[A](a: A): Task[A] = Value(Eval.now(a))
+
+  /**
+   * Construct a Task that when run will produce an A by evaluating
+   * the argument. This is an effect capturing constructor which is
+   * suitable for wrapping code which is side-effecting
+   */
+  def later[A](a: => A): Task[A] = Value(Eval.later(a))
+
+  /**
+   * Construct a task from a thunk that will be executed each time the
+   * Task is run. This is an effect capturing constructor which is
+   * suitable for wrapping code which is side-effecting.
+   */
+  def always[A](a: () => A): Task[A] = Value(new Always(a))
+
+  /**
+   *  Construct a Task which will represent an asynchronous
+   *  computation. The constructor takes a function which will be
+   *  invoked when the Task is run, passing the task a callback (A =>
+   *  Unit) which will be called when the asyncrounous computation is
+   *  complete.
+   */
+  def async[A](cb: (A => Unit) => Unit): Task[A] = Async(cb)
+
+  // Here we already have an eval, so we just have to return it
+  private[cats] final case class Value[A](eval: Eval[A]) extends Task[A] {
+    override def map[B](f: A => B): Task[B] = Value(eval.map(f))
+  }
+
+  // Store a flatMap operation on the Heap. when asked to eval, we
+  // flatMap against the given task, using Eval.defer to ensure that
+  // the computation is stack-safe
+  private[cats] final class Bind[Z, A](t: Task[Z], f: Z => Task[A]) extends Task[A] {
+    override def eval: Eval[A] = Eval.defer(t.eval.flatMap(f andThen (_.eval)))
+  }
+
+  /**
+   * Store the intent to catch non-fatal exceptions when running a Task.
+   */
+  private[cats] final case class AttemptNonFatal[A](task: Task[A]) extends Task[Xor[Throwable, A]] {
+    override def eval: Eval[Xor[Throwable, A]] =
+      Eval.always(Xor.catchNonFatal(task.eval.value))
+  }
+
+  /**
+   * Store the intent to catching some subset of the possible
+   * excpetions when running a Task.
+   */
+  private[cats] final case class AttemptOnly[T >: Null <: Throwable : ClassTag, A](task: Task[A]) extends Task[T Xor A] {
+    override def eval: Eval[Xor[T, A]] =
+      Eval.always(Xor.catchOnly[T](task.eval.value))
+  }
+
+  private[cats] final case class Async[A](complete: (A => Unit) => Unit) extends Task[A] {
+    override def eval: Eval[A] = {
+      val cdl = new CountDownLatch(1)
+      var result: Option[A] = None
+      complete((a: A) => {result = Some(a); cdl.countDown})
```

Comment:
`CountDownLatch` won't work in Javascript and besides, it's usage is a red flag on the JVM as well. There's absolutely no reason to block a thread in order to get an asynchronous value out of a `Task`.